server/core/include/scoped_client_identity.hpp

```diff
@@ -19,39 +20,74 @@ namespace irods::experimental
     class scoped_client_identity
     {
     public:
-        /// Constructs a scoped_client_identity object and sets the username of
-        /// rsComm_t::clientUser to new_username.
+        /// Constructs a scoped_client_identity object and updates the RsComm to represent
+        /// a user in the local zone.
         ///
-        /// \param[in] comm         The server communication object.
-        /// \param[in] new_username The name of the iRODS user to become.
+        /// The zone will match the string returned by getLocalZoneName().
         ///
-        /// \exception std::length_error Thrown if new_username exceeds the buffer size
-        ///                              of rsComm_t::clientUser::userName.
-        scoped_client_identity(rsComm_t& comm, std::string_view new_username)
-            : conn_{comm}
-            , username_{conn_.clientUser.userName}
+        /// \param[in] _comm         The server communication object.
+        /// \param[in] _new_username The name of the user to become.
+        ///
+        /// \exception irods::exception Thrown if _new_username is too large.
+        scoped_client_identity(RsComm& _comm, std::string _new_username)
+            : scoped_client_identity{_comm, _new_username, ""}
+        {
+        }
+
+        /// Constructs a scoped_client_identity object and updates the RsComm to represent
+        /// a user in a zone.
+        ///
+        /// \param[in] _comm         The server communication object.
+        /// \param[in] _new_username The name of the user to become.
+        /// \param[in] _new_zone     The name of the zone the user is a member of.
+        ///                          If _new_zone is empty, it defaults to the value returned
+        ///                          by getLocalZoneName().
+        ///
+        /// \exception irods::exception Thrown if _new_username or _new_zone is too large.
+        ///
+        /// \since 4.2.12
+        scoped_client_identity(RsComm& _comm, std::string _new_username, std::string _new_zone)
```

Comment:
Why not use `const std::string&` here? Seems very... intentional.

Reply:
It should be a const reference. I forgot to change it.

Will fix that right up.

Reply:
Done